### DISCUSSION TITLE
Cluster migration

### DISCUSSION THREAD

**Discussion by ddellarocca** [emqx/emqx#13262] [Q&A] [answered]
Hi,
We need to migrate from an EMQX cluster to another both deployed in K8S with the LoadBalancer service exposing the endpoints.
The idea is to change the DNS record pointing to the EMQX loadbalancer to the new one but we need a way to disconnect the clients from the old instance, is there a way to do that in batches? (we have more than 10k clients to disconnect and doing it in the UI is not feasible)
Thanks.

**Comment by zmstone** [comment, marked as answer]
Hi @ddellarocca 
For EMQX open-source edition, you can call batch-kickout API, doc here: 
https://docs.emqx.com/en/enterprise/v5.7/admin/api-docs.html#tag/Clients/paths/~1clients~1kickout~1bulk/post

For EMQX Enterprise, there is a nicer alternative: node evacuation.
https://docs.emqx.com/en/enterprise/latest/deploy/cluster/rebalancing.html#node-evacuation-and-cluster-load-rebalancing

EDIT: updated 1st link

**Reply by ddellarocca** [reply]
I see that is available since emqx 5.1, is there a way to do the same thing with emqx 5.0? (old cluster v5.0.11, new cluster v5.7)

**Reply by zmstone** [reply]
no, but for 10k clients, it should not take too long to enumerate all if you scriptize the process:
1. list client (one page)
2. kick the listed clients 
3. loop step 1

**Reply by ddellarocca** [reply]
I'll try that, thanks!
